controllers/controller.go

```diff
@@ -114,6 +114,10 @@ func (r *EtcdadmClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		log.Info("Cluster Controller has not yet set OwnerRef on etcd")
 		return ctrl.Result{}, nil
 	}
+	if !cluster.Status.InfrastructureReady {
```

Comment:
thanks for opening the PR, but instead of waiting and requeuing, we can change the Watch on Cluster object. For that we can change https://github.com/mrajashree/etcdadm-controller/blob/v1beta1/controllers/controller.go#L72 to use the CAPI predicate `ClusterUnpausedAndInfrastructureReady` instead of `ClusterUnpaused`

Reply:
Even with ClusterUnpausedAndInfrastructureReady, the reconcile function isn’t getting called after printing `Infrastructure cluster is not yet ready` unless I return the result with requeue or requestAfter.

Reply:
okay, not sure if this is an issue with stale objects in cache related to controller runtime, let's add both changes then. `ClusterUnpausedAndInfrastructureReady` and the requeue logic